src/main/java/com/thealgorithms/maths/VampireNumber.java

```diff
@@ -1,78 +1,65 @@
- * vampire number is 1260 = 21 × 60. *
+ * vampire number is 1260 = 21 × 60.
  *
- * <p>
- * link: https://en.wikipedia.org/wiki/Vampire_number *
- *
- * <p>
+ * @see <a href='https://en.wikipedia.org/wiki/Vampire_number'>Vampire number on Wikipedia</a>
  */
 public final class VampireNumber {
+    // Forbid instantiation.
     private VampireNumber() {
     }
 
-    public static void main(String[] args) {
-        test(10, 1000);
-    }
+    static boolean isVampireNumber(int a, int b, boolean ignorePseudoVampireNumbers) {
+        // Pseudo vampire numbers don't have to be of n/2 digits. E.g., 126 = 6 x 21 is such a number.
+        if (ignorePseudoVampireNumbers && String.valueOf(a).length() != String.valueOf(b).length()) {
+            return false;
+        }
 
-    static void test(int startValue, int stopValue) {
-        int countofRes = 1;
-        StringBuilder res = new StringBuilder();
+        String mulDigits = splitIntoSortedDigits(a * b);
+        String factorDigits = splitIntoSortedDigits(a, b);
 
-        for (int i = startValue; i <= stopValue; i++) {
-            for (int j = i; j <= stopValue; j++) {
-                // System.out.println(i+ " "+ j);
-                if (isVampireNumber(i, j, true)) {
-                    countofRes++;
-                    res.append("").append(countofRes).append(": = ( ").append(i).append(",").append(j).append(" = ").append(i * j).append(")").append("\n");
-                }
-            }
-        }
-        System.out.println(res);
+        return mulDigits.equals(factorDigits);
     }
 
-    static boolean isVampireNumber(int a, int b, boolean noPseudoVamireNumbers) {
-        // this is for pseudoVampireNumbers  pseudovampire number need not be of length n/2 digits
-        // for example 126 = 6 x 21
-        if (noPseudoVamireNumbers) {
-            if (a * 10 <= b || b * 10 <= a) {
-                return false;
+    // Method to split a pair of numbers to digits and sort them in the ascending order.
+    static String splitIntoSortedDigits(int... nums) {
+        // Collect all digits in a list.
+        ArrayList<Integer> digits = new ArrayList<>();
+        for (int num : nums) {
+            while (num > 0) {
+                digits.add(num % 10);
+                num /= 10;
             }
         }
 
-        String mulDigits = splitIntoDigits(a * b, 0);
-        String faktorDigits = splitIntoDigits(a, b);
-
-        return mulDigits.equals(faktorDigits);
+        // Sort all digits and convert to String.
+        StringBuilder res = new StringBuilder();
+        digits.stream().sorted().forEach(res::append);
+        return res.toString();
     }
 
-    // methode to Split the numbers to Digits
-    static String splitIntoDigits(int num, int num2) {
-        StringBuilder res = new StringBuilder();
+    static void printVampireNumbers(int startValue, int stopValue, boolean ignorePseudoVampireNumbers) {
```

Comment:
Please remove main and printVampireNumbers, we don't usually add visualization, only algorithms